qiskit/circuit/quantumcircuit.py

```diff
@@ -402,6 +404,65 @@ def from_qasm_str(qasm_str):
         return _circuit_from_qasm(qasm)
 
 
+def _1q_gate(func):
```

Comment:
can you move these decorators to a separate file `qiskit/circuit/decorators.py`? We have a similar file in `qiskit/test/`